libminifi/include/core/state/nodes/AgentInformation.h

```diff
@@ -623,18 +632,70 @@ class AgentManifest : public DeviceInformation {
 };
 
 /**
- * Purpose and Justification: Prints classes along with their properties for the current agent.
+ * This class is used for regular heartbeat without manifest
+ * A light weight heartbeat
  */
-class AgentInformation : public DeviceInformation, public AgentMonitor, public AgentIdentifier {
+class AgentInformationWithoutManifest : public DeviceInformation, public AgentMonitor, public AgentIdentifier {
```

Comment:
Public inheritance means is-a relationship. By looking at the names, this class fits none of its base classes.
Use composition (preferred) or private inheritance for implemented-in-terms-of relationship.

Couldn't this class become just a `serialize_agent_information_without_manifest(...)` free function? OOP is a toolbox, not an obligation.

Reply:
Did not understand why it does not fit is-a relationship to any of its base classes. Can you give example?

No this cannot become a free function unless we change the way we register and load classes. 
This instantiation of this class and other class (AgentInformation) depends on configuration of root node classes in "nifi.c2.root.classes" property that are loaded during C2 initialization in FlowController. I want to keep it that way and be consistent with the existing code base.

Reply:
"agent information (without manifest) is a device information"
"agent information (without manifest) is an agent monitor"
"agent information (without manifest) in an agent identifier"
None of the above statements sound true, although I'm not sure about the first one.

In the meantime I realized that this relationship was not introduced now but taken over from old `AgentInformation`, so I'm fine with not changing the design.

Thanks for the explanation why it cannot be a free function. Feel free to mark this resolved.

Reply:
I just happened to see this E-mail and I think this is spot on. This evolved poorly and AgentInformation became more of a compositional relationship with these other artifacts as they changed in name and purpose. 

This should be improved at some point for maintainability.